irohad/torii/impl/command_service.cpp

```diff
@@ -130,4 +136,117 @@ namespace torii {
+                  response_writer.WriteLast(resp_sub, grpc::WriteOptions());
+                  subscription.unsubscribe();
+                  finished = true;
+                  cv_.notify_one();
+                  break;
+                case iroha::model::TransactionResponse::
+                    STATELESS_VALIDATION_SUCCESS:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::STATELESS_VALIDATION_SUCCESS);
+                  response_writer.Write(resp_sub);
+                  break;
+                case iroha::model::TransactionResponse::
+                    STATEFUL_VALIDATION_FAILED:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::STATEFUL_VALIDATION_FAILED);
+                  response_writer.WriteLast(resp_sub, grpc::WriteOptions());
+                  subscription.unsubscribe();
+                  finished = true;
+                  cv_.notify_one();
+                  break;
+                case iroha::model::TransactionResponse::
+                    STATEFUL_VALIDATION_SUCCESS:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::STATEFUL_VALIDATION_SUCCESS);
+                  response_writer.Write(resp_sub);
+                  break;
+                case iroha::model::TransactionResponse::COMMITTED:
+                  resp_sub.set_tx_status(iroha::protocol::TxStatus::COMMITTED);
+                  response_writer.WriteLast(resp_sub, grpc::WriteOptions());
+                  subscription.unsubscribe();
+                  finished = true;
+                  cv_.notify_one();
+                  break;
+                case iroha::model::TransactionResponse::IN_PROGRESS:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::IN_PROGRESS);
+                  response_writer.Write(resp_sub);
+                  break;
+                case iroha::model::TransactionResponse::NOT_RECEIVED:
+                default:
+                  resp_sub.set_tx_status(
+                      iroha::protocol::TxStatus::NOT_RECEIVED);
+                  response_writer.WriteLast(resp_sub, grpc::WriteOptions());
+                  subscription.unsubscribe();
+                  finished = true;
+                  cv_.notify_one();
+                  break;
+              }
+            });
+
+    std::unique_lock<std::mutex> lock(wait_subscription_);
+    // One second should be enough to at least start tx processing.
+    // Otherwise we think there is no such tx at all.
+    cv_.wait_for(lock, 1s);
+    if (not finished) {
+      if (not resp) {
+        subscription.unsubscribe();
+        iroha::protocol::ToriiResponse resp_none;
+        resp_none.set_tx_hash(request_hash);
+        resp_none.set_tx_status(iroha::protocol::TxStatus::NOT_RECEIVED);
+        response_writer.WriteLast(resp_none, grpc::WriteOptions());
+      } else {
+        cv_.wait_for(lock, 10s);
```

Comment:
Same about class dependency. But in this case, we may say, that we require the time of one proposal propagation as a parameter.